### ISSUE TITLE
mask function hangs on warning

### ISSUE THREAD

**Issue by luraess** [rafaqz/Rasters.jl#344]
Since this https://github.com/rafaqz/Rasters.jl/issues/341#issuecomment-1332827746 was fixed, I am now getting an "error" in a [`mask_trim` function](ttps://github.com/eth-vaw-glaciology/GlacioTools.jl/blob/8ad0e68bfe04fbb1782afa442256adba06264846/src/Rasters_helpers.jl#L38) used here https://github.com/eth-vaw-glaciology/GlacioTools.jl/blob/8ad0e68bfe04fbb1782afa442256adba06264846/src/Alpine_glaciers.jl#L87

It seems that masking fails on the raster / polygon combination I am trying to use.

Making a MWE for this not straight fwd, but the polygon as well as error stack is attached here:
- [polygon.txt](https://github.com/rafaqz/Rasters.jl/files/10141757/polygon.txt)
- warning that spins forever [error.txt](https://github.com/rafaqz/Rasters.jl/files/10141761/error.txt)

**Comment by rafaqz** [comment]
Can you post your code and errors inline in code blocks? I can't read it easily without highlighting, especially on my phone where I often answer these and seeing it immediately in the app removes 90% of the effort.

Guess I need to put this in the issue obstructions.

**Comment by luraess** [comment]
This seems to fix the issue https://github.com/eth-vaw-glaciology/GlacioTools.jl/blob/896f5466621f631c13b5fadaf76d8a19d9aa00e1/src/Rasters_helpers.jl#L51-L58

**Comment by mauro3** [comment]
Above circumvents the issue. 

Maybe of note is that the shapefile is a 3D polygon.  Here a stacktrace of the warning (which I turned into an error using the trick in https://discourse.julialang.org/t/terminate-after-warning-is-thrown-global-setting/80368):
```julia
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.3 (2022-11-14)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |

julia> using GlacioTools

julia> datadir = joinpath(@__DIR__,"../mydata/")
"~/.julia/dev/GlacioTools/../mydata/"

julia> redirect_stderr(open(touch(tempname()), "r"))
IOStream(<file /tmp/jl_JUzGHI5gvS>)

julia> SGI_ID = "B43-03"
"B43-03"

julia> data = fetch_glacier("Rhone", SGI_ID; datadir)
Reading in global data... done.
ERROR: ArgumentError: write failed, IOStream is not writeable
Stacktrace:
  [1] unsafe_write(s::IOStream, p::Ptr{UInt8}, nb::UInt64)
    @ Base ./iostream.jl:373
  [2] unsafe_write
    @ ./io.jl:683 [inlined]
  [3] write(s::IOStream, a::Vector{UInt8})
    @ Base ./io.jl:706
  [4] handle_message(logger::Logging.ConsoleLogger, level::Base.CoreLogging.LogLevel, message::Any, _module::Any, group::Any, id::Any, filepath::Any, line::Any; kwargs::Base.Pairs{Symbol, V, Tuple{Vararg{Symbol, N}}, NamedTuple{names, T}} where {V, N, names, T<:Tuple{Vararg{Any, N}}})
    @ Logging ~/julia/julia-1.8/usr/share/julia/stdlib/v1.8/Logging/src/ConsoleLogger.jl:178
  [5] handle_message(logger::Logging.ConsoleLogger, level::Base.CoreLogging.LogLevel, message::Any, _module::Any, group::Any, id::Any, filepath::Any, line::Any)
    @ Logging ~/julia/julia-1.8/usr/share/julia/stdlib/v1.8/Logging/src/ConsoleLogger.jl:106
  [6] macro expansion
    @ ./logging.jl:365 [inlined]
  [7] _warnextradims(extradims::Tuple{Rasters.Band{Int64}})
    @ DimensionalData.Dimensions ~/.julia/packages/DimensionalData/K9D4P/src/Dimensions/primitives.jl:645
  [8] dims2indices
    @ ~/.julia/packages/DimensionalData/K9D4P/src/Dimensions/indexing.jl:48 [inlined]
  [9] dims2indices
    @ ~/.julia/packages/DimensionalData/K9D4P/src/Dimensions/indexing.jl:34 [inlined]
 [10] #getindex#39
    @ ~/.julia/packages/DimensionalData/K9D4P/src/array/indexing.jl:36 [inlined]
 [11] getindex
    @ ~/.julia/packages/DimensionalData/K9D4P/src/array/indexing.jl:36 [inlined]
 [12] _inner_fill_polygon!(B::Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, BitArray{3}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Missing}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Bool}, geom::Shapefile.PolygonZ, inpoly::SubArray{Bool, 1, Matrix{Bool}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, reshaped::Rasters.Raster{Bool, 2, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}}, Tuple{}, Base.ReshapedArray{Bool, 2, SubArray{Bool, 1, Matrix{Bool}, Tuple{Base.Slice{Base.OneTo{Int64}}, Int64}, true}, Tuple{}}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Missing}, shifted_dims::Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, Vector{Float64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Center}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, Vector{Float64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Center}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}}; fill::Bool, boundary::Symbol, kw::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Rasters ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:104
 [13] _fill_polygon!(B::Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, BitArray{3}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Missing}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Bool}, geom::Shapefile.PolygonZ; fill::Bool, boundary::Symbol, geomextent::Extents.Extent{(:X, :Y, :Z), Tuple{Tuple{Float64, Float64}, Tuple{Float64, Float64}, Tuple{Float64, Float64}}}, kw::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:shape, :to), Tuple{Symbol, Rasters.Raster{Float32, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Array{Float32, 3}, Symbol, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, Float32}}}})
    @ Rasters ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:95
 [14] _fill_geometry!(B::Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, BitArray{3}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Missing}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Bool}, ::GeoInterface.MultiPolygonTrait, geom::Shapefile.PolygonZ; shape::Nothing, kw::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:fill, :to), Tuple{Bool, Rasters.Raster{Float32, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Array{Float32, 3}, Symbol, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, Float32}}}})
    @ Rasters ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:36
 [15] #_fill_geometry!#159
    @ ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:15 [inlined]
 [16] _fill_geometry!(B::Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Rasters.Raster{Bool, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, BitArray{3}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Missing}, Symbol, DimensionalData.Dimensions.LookupArrays.NoMetadata, Bool}, trait::Nothing, geoms::SubArray{Union{Missing, Shapefile.PolygonZ}, 1, Vector{Union{Missing, Shapefile.PolygonZ}}, Tuple{Vector{Int64}}, false}; kw::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:fill, :to), Tuple{Bool, Rasters.Raster{Float32, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Array{Float32, 3}, Symbol, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, Float32}}}})
    @ Rasters ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:44
 [17] #_fill_geometry!#159
    @ ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:15 [inlined]
 [18] #fill_geometry!#158
    @ ~/.julia/packages/Rasters/3WVZm/src/polygon_ops.jl:11 [inlined]
 [19] #boolmask!#455
    @ ~/.julia/packages/Rasters/3WVZm/src/methods/mask.jl:270 [inlined]
 [20] boolmask(x::SubArray{Union{Missing, Shapefile.PolygonZ}, 1, Vector{Union{Missing, Shapefile.PolygonZ}}, Tuple{Vector{Int64}}, false}; kw::Base.Pairs{Symbol, Rasters.Raster{Float32, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Array{Float32, 3}, Symbol, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, Float32}, Tuple{Symbol}, NamedTuple{(:to,), Tuple{Rasters.Raster{Float32, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Array{Float32, 3}, Symbol, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, Float32}}}})
    @ Rasters ~/.julia/packages/Rasters/3WVZm/src/methods/mask.jl:261
 [21] #_mask#430
    @ ~/.julia/packages/Rasters/3WVZm/src/methods/mask.jl:102 [inlined]
 [22] _mask
    @ ~/.julia/packages/Rasters/3WVZm/src/methods/mask.jl:101 [inlined]
 [23] #mask#428
    @ ~/.julia/packages/Rasters/3WVZm/src/methods/mask.jl:94 [inlined]
 [24] mask_trim(rasterDat::Rasters.Raster{Float32, 3, Tuple{DimensionalData.Dimensions.X{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.X{Colon}}}, DimensionalData.Dimensions.Y{Rasters.Projected{Float64, LinRange{Float64, Int64}, DimensionalData.Dimensions.LookupArrays.ReverseOrdered, DimensionalData.Dimensions.LookupArrays.Regular{Float64}, DimensionalData.Dimensions.LookupArrays.Intervals{DimensionalData.Dimensions.LookupArrays.Start}, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, GeoFormatTypes.WellKnownText{GeoFormatTypes.CRS}, Nothing, DimensionalData.Dimensions.Y{Colon}}}, Rasters.Band{DimensionalData.Dimensions.LookupArrays.Categorical{Int64, UnitRange{Int64}, DimensionalData.Dimensions.LookupArrays.ForwardOrdered, DimensionalData.Dimensions.LookupArrays.NoMetadata}}}, Tuple{}, Array{Float32, 3}, Symbol, DimensionalData.Dimensions.LookupArrays.Metadata{Rasters.GDALfile, Dict{String, Any}}, Float32}, poly::SubArray{Union{Missing, Shapefile.PolygonZ}, 1, Vector{Union{Missing, Shapefile.PolygonZ}}, Tuple{Vector{Int64}}, false}, pad::Int64)
    @ GlacioTools ~/julia/dot-julia-dev/GlacioTools/src/Rasters_helpers.jl:38
 [25] geom_select(SGI_ID::String, name::String, datadir::String; padding::Int64, do_save::Bool)
    @ GlacioTools ~/julia/dot-julia-dev/GlacioTools/src/Alpine_glaciers.jl:87
 [26] geom_select
    @ ~/julia/dot-julia-dev/GlacioTools/src/Alpine_glaciers.jl:61 [inlined]
 [27] fetch_glacier(name::String, SGI_ID::String; datadir::String)
    @ GlacioTools ~/julia/dot-julia-dev/GlacioTools/src/Alpine_glaciers.jl:30
 [28] top-level scope
    @ REPL[6]:1

julia>
```

**Comment by rafaqz** [comment]
@luraess I in no way meant that I wouldn't fix this, so reopening. 

@mauro3 thanks for the MWE

**Comment by rafaqz** [comment]
I think this is solved by #335 already, I'm just having a bunch of unrelated problems getting CI passing so it hasn't been merged.

**Comment by rafaqz** [comment]
This should be fixed on master now.
